src/onedrive.d

```diff
@@ -10,11 +10,14 @@ private immutable {
 	string authUrl = "https://login.microsoftonline.com/common/oauth2/v2.0/authorize";
 	string redirectUrl = "https://login.microsoftonline.com/common/oauth2/nativeclient";
 	string tokenUrl = "https://login.microsoftonline.com/common/oauth2/v2.0/token";
-	string driveUrl = "https://graph.microsoft.com/v1.0/me/drive";
 	string itemByIdUrl = "https://graph.microsoft.com/v1.0/me/drive/items/";
```

Comment:
Do none of these other URLs need to change?

Reply:
Everything, except the driveURL - which is only used initially to get the driveId. The only suspect: itemByIdUrl, is never used in the code.

Reply:
Might be best to remove that then.

Reply:
Well, I only found this out after you asked. Also, I would rather hesitate to change code unrelated to my patch.

Reply:
Yes, but if `itemByIdUrl` was later used in the codebase after your changes were applied when a user had configured a custom drive it would cause breakage.

You should either:
1. Remove `itemByIdUrl` from the codebase (my preference), or
2. Ensure that `itemByIdUrl` will always be set to a value compatible with `driveUrl`.

Reply:
You convinced me. I did 1. and found a piece of related cruft code: iewChangesByPath(), which is also not used and is probably even incorrect.

Commit follows soon.

Reply:
Awesome! Always nice when you can leave a codebase cleaner than when you started!

Reply:
Mhm.. but I messed up again with git by including (unintentionally) the other patch...

Reply:
Best to create a feature branch, then cherry-pick the commits you want into that for this PR.

Reply:
Its more complicated. I did this, but somehow merged in the changes from the other branch.
Is it ok to rebase the branch?

Reply:
I'd love to see this in action, but I don't understand how to use the instruction to get the drive_id. 

Do you plan to add some way of automating this?

Reply:
Yes, sorry, the instruction are to scarce to use without further knowledge.

1. go to the Microsoft Graph API Explorer: https://developer.microsoft.com/en-us/graph/graph-explorer
2. Login in to your Microsoft Account with the respective button in the menu on the left side.
3. Construct the URL as shown in the README.odmon.md file, paste it into the url input line and press the "Run Query" button.

You should get a JSON formatted response which contains the drive_id as "id:".

I am currently putting together a script for managing several onedrive processes running on diferents drives, and for extracting the drive_id.  It's written in fairly basic Tcl/Tk.

Reply:
Great, this seems to work perfectly now. 

I'd love to try out your script. I'm usually a python/QT kindof person but I'd be glad to help if I can.

Reply:
I have rounded up the script and published it to http://at.magma-soft.at/darcs/odmon/. You can add a drive for syncronization by pasting the odopen:// url directly to the odmon application. Please read the docs.

Reply:
Thanks!

I tried it out and it looks promising but I ran into some troubles. Would like to continue the discussion elsewhere or would you rather have me post here?

In short I managed to sync my personal drive but when adding a new business drive I got an error "key "access_token" not known in dictionary". Before that it had managed to retrieve a bunch of stuff and it looked like it got the drive_id correctly.

Reply:
Feel free to use the Email address in the README.md file of odmon to contact me directly. This problem is not related to onedrive.

If you got the drive_id in the log output, you can set up syncing manually by copying it into the respective config file: `drive_id = "<drive_id from odmon log>"`.